openadapt/ml/openai_finetune.py

```diff
@@ -0,0 +1,124 @@
+from openadapt.crud import *
+from openadapt import models, utils
+from copy import deepcopy
+from loguru import logger
+import json
+import openai
+import sys
+
+
+from openadapt.strategies.stateful import get_window_state_diffs
+
+
+def condense_window_state(recording_id: int):
+    """
+    Takes in a recording as input and outputs a list of tuples
+    of action, window events where window events are condensed
+    to contain basic information, as in stateful_test.py
+    """
+    grab_recording = get_recording_by_id(recording_id)
+    total_acx = get_action_events(grab_recording)
+
+    for i in range(len(total_acx)):
+        processed_acx, processed_wd = (
+            sanitize(total_acx[i])[0],
+            sanitize(total_acx[i])[1],
+        )
+        processed_wd.pop("meta")
+        # task_description = recording.task_description
+
+        logger.debug(f"{total_acx[i].timestamp=}")
+        logger.debug(f"{total_acx[i].window_event_timestamp=}")
+        # which timestamp to finetune on?
+        total_acx[i] = (processed_acx, processed_wd)
+    return total_acx
+
+
+def write_to_file(recording_id: int):
```

Comment:
I've thought of just renaming this function "openai_dataset_prep" because other models have different formats that they require the dataset to be in.

Reply:
I agree that we likely need some generic interface -- what do you think about calling it `persist_finetune_data` or similar?